### PR TITLE
circuitv2: correctly set the transport in the ConnectionState

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

File: p2p/protocol/circuitv2/client/conn.go [hunk 1]
```diff
@@ -143,3 +144,20 @@ func (c *Conn) untagHop() {
...
+	network.ConnStat
+}
+
+type capableConn struct {
+	capableConnWithStat
+}
+
+var transportName = ma.ProtocolWithCode(ma.P_CIRCUIT).Name
+
+func (c *capableConn) ConnState() network.ConnectionState {
```
Comment (MarcoPolo) [inline p2p/protocol/circuitv2/client/conn.go:159]: Should we make this a value receiver instead? That lets us avoid pointer chasing.

**Review by MarcoPolo — APPROVED** [review]

File: p2p/protocol/circuitv2/client/conn.go (hunk 1 above)
Comment (marten-seemann) [reply p2p/protocol/circuitv2/client/conn.go]: Done.

File: p2p/protocol/circuitv2/client/transport.go [hunk 2]
```diff
@@ -63,7 +63,11 @@ func (c *Client) Dial(ctx context.Context, a ma.Multiaddr, p peer.ID) (transport
 		return nil, err
 	}
 	conn.tagHop()
-	return c.upgrader.Upgrade(ctx, c, conn, network.DirOutbound, p, connScope)
+	cc, err := c.upgrader.Upgrade(ctx, c, conn, network.DirOutbound, p, connScope)
+	if err != nil {
+		return nil, err
+	}
+	return &capableConn{cc.(capableConnWithStat)}, nil
```
Comment (MarcoPolo) [inline p2p/protocol/circuitv2/client/transport.go:70]:
```suggestion
	return capableConn{cc.(capableConnWithStat)}, nil
```

**Review by MarcoPolo — CHANGES_REQUESTED** [review]
I don't think you need to create a pointer there

**Review by MarcoPolo — APPROVED** [review]
sorry I meant to approve this with the suggestion.

File: p2p/protocol/circuitv2/client/transport.go (hunk 2 above)
Comment (marten-seemann) [reply p2p/protocol/circuitv2/client/transport.go]: Done.

**Pull request merged**